source/parsed_grid_generator.cc

```diff
@@ -290,8 +267,60 @@ void ParsedGridGenerator<dim, spacedim>::create(Triangulation<dim,spacedim> &tri
 
 }
 
+template <int dim>
+void ParsedGridGenerator<dim,dim>::create(Triangulation<dim,dim> &tria)
+{
+  Assert(this->grid_name != "", ExcNotInitialized());
+  if (this->grid_name == "rectangle")
+    {
+      GridGenerator::subdivided_hyper_rectangle ( tria,
+                                                  this->un_int_vec_option_one,
+                                                  this->point_option_two,
+                                                  this->point_option_one,
+                                                  this->bool_option_one);
+    }
+  else if (this->grid_name == "unit_hyperball")
+    {
+      Point<dim> center;
+      double radius=1.;
+      GridGenerator::hyper_ball (tria, center, radius);
+    }
+  else if (this->grid_name == "subhyperrectangle")
+    {
+      std::vector<unsigned int> refinement(dim);
+      for (unsigned int i=0; i<dim; ++i)
+        refinement[i] = this->un_int_option_one;
+      GridGenerator::subdivided_hyper_rectangle ( tria,
+                                                  refinement,
+                                                  this->point_option_two,
+                                                  this->point_option_one);
+    }
```

Comment:
È lo stesso ma con la vecchia implementazione non te lo faceva fare perché il compilatore trovata tria definita con dim e spacedim differenti e ti dava errore. Separando i casi in due classi figlie ho aggirato il problema.